traffic_ops/traffic_ops_golang/deliveryservice/request/requests.go

```diff
@@ -383,6 +401,20 @@ func (req *deliveryServiceRequestAssignment) Update() (error, error, int) {
 	req.DeliveryServiceRequestNullable = current.DeliveryServiceRequestNullable
 	req.AssigneeID = assigneeID
 
+	err, existingLastUpdated := CheckIfExistsBeforeUpdate(req.ReqInfo.Tx, *req.ID)
+	if err != nil {
+		return errors.New("no delivery service request found with this id"), nil, http.StatusNotFound
+	}
+	_, iumsTime := rfc.GetETagOrIfUnmodifiedSinceTime(h)
+	existingEtag := rfc.ETag(existingLastUpdated.Time)
+
+	if h.Get(rfc.IfMatch) != "" && !strings.Contains(h.Get(rfc.IfMatch), existingEtag) {
+		return errors.New("header preconditions dont match"), nil, http.StatusPreconditionFailed
```

Comment:
The text here doesn't have to match the code, and since it's a user error, I think it's safe (from a security perspective) to let them know why they're getting a 412. Could we make the message something more informative like "resource was modified" ?